Add support for loading mods on Android

BeeeWall opened the pull request:
What I did was set it to copy mods from `/sdcard/Android/data/com.unciv.app/files/mods` to the internal app directory (where libGDX reads files from) every launch. I didn't change anything beyond that (yet), as that was enough to get the example mods to load. 

May not be the best to copy them every launch like I'm doing, but short of replacing `Gdx.files.local` with a function checking that AND the external directory, not sure how else to make it use a location the user can write to.

yairm210 commented:
In general I'm against adding any kind of permission which is not absolutely necessary
I'm not sure how many people, if at all, actually use mods, and how many would do so if this was enabled
I personally haven't seen any completed mods apart from my example one so I feel this is not really something that's very common

BeeeWall commented:
The extra permission is *only* for devices below Android 4.4, which is probably a very small portion of users (3.5% of Android users use OS versions above 4.0 and below 4.4, although I don't know if that is different for this app). It doesn't even request the permission on any newer device. 

I haven't seen a whole lot of mods either, but I feel like supporting them in case some get made isn't a bad idea (obviously your call as the dev, though). 

Hopefully there's a better way than copying the mods every startup like I'm doing it here though (if there's a way to get everything stored in the external app storage instead of internal, that might actually be good, because then users can copy their saves, but I'm not sure how to do so, and that's a separate issue I think).

BeeeWall commented:
I have removed the permission and replaced it with a version check. On Android 4.4+, mods will be copied since the permission isn't needed. On older versions, it will simply skip that step.

(I moved the mod copying to its own function because I personally feel it's nicer with the if statement that way, but that's personal opinion and I can change it if you'd rather that. Not sure what your opinion on that sort of thing is.)

yairm210 commented:
Okay, looks good!
Did you test this with an example mod?

BeeeWall commented:
Yep, tested it with Inca and Snow. They both loaded and worked as well as they did on a PC (as in, sorta buggy but functional). 

Which reminds me, the Snow mod crashes the game after a few turns (not just my version, but the normal as well), unless a commit in the past few days has fixed it. But that's a seperate issue that I'll try and file later when I've got time. Iirc it was something about no improvements found for snow, but I'll need to check and post a log.
